### PR TITLE
Bmc itsm integration

### PR DESCRIPTION
**Pull request by OfekMQmasters**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
BMC Helix integration
## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ X] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [X ] Tests
- [X ] Documentation

### CODE REVIEW

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (JudahSchwartz) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:2348]: please be consistent with the casing

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (JudahSchwartz) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:2519]: here too. For example "known error" is lowercase while its upercase in the argument above

**Review by JudahSchwartz — COMMENTED** [review]
Looks very good.
Please use args.get instead of direct access,
Please be consistent about periods/casing in the docs/yml

**Comment by julieschwartz18** [comment]
Hi @JudahSchwartz @idovandijk @AradCarmi - I do not have permissions to edit the yaml files, can you provide that? Thanks

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:24]: I recommend "Maximum number of incidents per fetch."

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:54]: I recommend "The status of the tickets to fetch. Since each ticket type has its own unique status pool, consider selecting only those who match the selected ticket types."

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:106]: Need to correct the spelling to **urgency**

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:118]: Need to fix capitalization. Should be:
Search qualification to fetch tickets. For example: Status = "Draft" AND Impact = "1-Extensive/Widespread". The query is used as an addition to the existing params. See the BMC documentation for how to build search qualifications: https://docs.bmc.com/docs/ars2008/building-qualifications-and-expressions-929630007.html'

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:123]: Need to add Cortex to the term XSOAR:
Choose the direction to mirror the incident: Incoming (from  BMC Helix ITSM to Cortex XSOAR), Outgoing (from Cortex XSOAR to BMC Helix ITSM), or Incoming and Outgoing (from/to Cortex XSOAR and BMC Helix ITSM).

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:136]: Need to ad Cortex to the term XSOAR:
Close Mirrored Cortex XSOAR Incident

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:140]: Need to add Cortex to the term XSOAR:
When selected, closing the Cortex XSOAR incident is mirrored in BMC Helix ITSM.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:169]: Recommend the following:
List of user profiles in BMC Helix ITSM. The records are retrieved by the query argument or by the filtering arguments. When using filtering arguments, Each one defines a 'LIKE' operation, and the 'AND' operator is used between them.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:172]: The user ID.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:175]: The user first name.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:178]: The user last name.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:182]: A comma-separated list of user IDs. Used as a filtering argument.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:189]: The search qualification. For example: Status = "Draft" AND Impact = "1-Extensive/Widespread". The query is used in addition to the existing arguments. See BMC documentation for how to Build search qualifications: https://docs.bmc.com/docs/ars2008/building-qualifications-and-expressions-929630007.html'

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:217]: The user first name. Used as a filtering argument.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:224]: The user last name. Used as a filtering argument.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:231]: The user company name. Used as a filtering argument.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:238]: The user department name. Used as a filtering argument.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:245]: The user organization name. Used as a filtering argument.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:252]: Lists companies in BMC Helix ITSM. The records are retrieved by the query argument or by the filtering arguments. When using filtering arguments, each one defines a 'LIKE' operation and the 'AND' operator is used between them.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:256]: The company ID.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:259]: The company name.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.yml:262]: The company type.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM_description.md
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM_description.md:4]: The user role should have access to service request, change request, problem investigation, known error, task, and incident tickets.
**Note:**
Only the Admin role can delete a ticket.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM_description.md
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM_description.md:7]: - **Display ID** (ID number) is a unique ID that references a specific ticket. This field is comparable to the ticket ID as it displayed in BMC Helix ITSM.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM_description.md
Comment (julieschwartz18) [inline Packs/BmcITSM/Integrations/BmcITSM/BmcITSM_description.md:9]: - **Request ID** is a unique core field of the ticket. Although it is not exposed to the user, it directly accesses the resource ticket through REST API calls, and executes Update, Delete, and Get operations on the ticket. In the integration it is a required argument for most of the commands.

File: Packs/BmcITSM/README.md
Comment (julieschwartz18) [inline Packs/BmcITSM/README.md:4]: - Views, creates, updates, or deletes a BMC Helix ITSM ticket directly from Cortex XSOAR.

File: Packs/BmcITSM/README.md
Comment (julieschwartz18) [inline Packs/BmcITSM/README.md:7]: - Mirrors incoming BMC Helix ITSM tickets to Cortex XSOAR.

File: Packs/BmcITSM/README.md
Comment (julieschwartz18) [inline Packs/BmcITSM/README.md:8]: - Mirrors out tickets from Cortex XSOAR to BMC Helix ITSM.

File: Packs/BmcITSM/README.md
Comment (julieschwartz18) [inline Packs/BmcITSM/README.md:5]: - Views, creates, updates, and deletes records from BMC Helix ITSM tables: Users, Companies, and Ticket templates.

File: Packs/BmcITSM/README.md
Comment (julieschwartz18) [inline Packs/BmcITSM/README.md:6]: - Fetches BMC Helix ITSM tickets.

File: Packs/BmcITSM/README.md
Comment (julieschwartz18) [inline Packs/BmcITSM/README.md:9]: - Supports service request, incident, change request, task, problem investigation and known error ticket types.

**Comment by OfekMQmasters** [comment]
Hey @julieschwartz18 , 
Now You have permissions to edit the BmcITSM.yml file, so you can make your modifications. 
@JudahSchwartz FYI

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.py
Comment (OfekMQmasters) [reply Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.py:235]: Done.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.py
Comment (OfekMQmasters) [reply Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.py:2636]: Fixed.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.py
Comment (OfekMQmasters) [reply Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.py:2918]: Fixed.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.py
Comment (OfekMQmasters) [reply Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.py:3513]: Added an API call.

File: Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.py
Comment (OfekMQmasters) [reply Packs/BmcITSM/Integrations/BmcITSM/BmcITSM.py:1332]: Added relevant message.

File: Tests/conf.json
Comment (JudahSchwartz) [reply Tests/conf.json:5443]: @OfekMQmasters this looks like its still an issue. It probably is why the demisto-sdk is failing on parsing json

**Comment by julieschwartz18** [comment]
Doc review is complete for the BmcITSM.yml file. This commit may have overwritten earlier updates, please review and confirm (sorry about that!)

**Comment by julieschwartz18** [comment]
Doc review complete.

**Review by idovandijk — COMMENTED** [review]
Hi, thank you for the contribution!
Some files, such as the classifier, still need to be formatted using the Demisto SDK `format` command, otherwise they may not be usable / viewable for my review.
A few more comments:
1. A strong suggestion - you may want to change the incident field of "BMC Assignee" into an indicator type, which has the indicator fields of "Full Name", "Group" and "Support Organization", "Support Company", rather than having the BMC Assignee a grid incident field with those columns.
That would be the more correct way to do it, it would allow doing things like searching in XSOAR for a BMC assignee who is part of a specific group, for example.
Should you decide to take my suggestion - note that fields that may be applicable to other products/packs, should be added to the Common packs (in this case - CommonTypes pack). For example the Full Name field probably already exists, or may very well be added there :)
2. Should BMC Contact Sensitivity be changed to a single-select field? Hint: if the optional items are ever to be changed, the answer would be no.
3. BMC Customer field, BMC Requester, other grid fields which behave similarly - same as my 1st point.
4. Risk Level field - same as point 2.

Finally - since this PR consists of classifier/mapper, incident fields and a layout, I would be happy for a short demo showing those in the flow of data from ingestion to incident. Can you share with me some time windows please?

**Review by idovandijk — COMMENTED** [review]
Hi @OfekMQmasters, awesome work.
I've watched the detailed demo videos you've provided to us. I have 2 more points:
- When listing users - additional properties might be better as a flag, as it could easily clutter the context.
- IIRC command descriptions support Markdown. You can use that to link to documentation resources in a more user friendly manner.

In this PR there are no mandatory fixes from my side. Also as you've already sent us demo videos, I don't see the need in a live demo. Feel free to address my current and previous points and makes changes as you desire.
@JudahSchwartz if there are no code changes required, we can go ahead and you can do the approval.

**Review by JudahSchwartz — APPROVED** [review]
Great! looks good to me. Please wait for @julieschwartz18 and @idovandijk 's approval before merge

**Review by idovandijk — APPROVED** [review]

**Pull request merged**
